### PR TITLE
Make Sipa HSS again

### PR DESCRIPTION
**Pull request by Lodifice**
HSS passive members can see their traffic volume and history now.
HSS members can change the MAC address by themselves.
HSS members can activate cache usage.

No tests included.

### CODE REVIEW

**Comment by toothstone** [comment]
I don't think it was a good idea to include the cache stuff in this PR, now this will have to wait until the hades implementation for HSS caching is done as well.

**Comment by Lodifice** [comment]
I don't think we have to wait on that, since our news and cache info page state that it is not working in HSS currently.

If some one logs in for no reason and finds out that it is suddenly possible to "activate" cache usage and just does it without reading the information, that is not our problem imho.

**Comment by lukasjuhrich** [comment]
@Lodifice I agree with @MarauderXtreme and don't think we should publish it atm as this could be a silent cause of undesired behavior. I will revert this commit so it remains in the history and can be reintroduced at the right time.

File: sipa/model/hss/user.py [hunk 1]
```diff
@@ -223,7 +223,19 @@ def login(self):
 
     @active_prop
     def mac(self):
-        return ", ".join(mac.mac.lower() for mac in self._pg_account.macs)
+        return {'value': ", ".join(mac.mac.lower() for mac in self._pg_account.macs),
+                'tmp_readonly': len(self._pg_account.macs) > 1}
```
Comment (lukasjuhrich) [inline sipa/model/hss/user.py]: Good catch with `tmp_readonly`!

File: sipa/model/hss/user.py [hunk 2]
```diff
@@ -241,7 +253,7 @@ def address(self):
 
     @active_prop
     def status(self):
-        if self.has_connection:
+        if self._pg_account.properties.active:
```
Comment (lukasjuhrich) [inline sipa/model/hss/user.py]: Thanks for fixing #271 as well.

File: sipa/model/hss/user.py [hunk 3]
```diff
@@ -271,7 +283,7 @@ def userdb(self):
 
     @property
     def has_connection(self):
-        return self._pg_account.properties.active
+        return True
```
Comment (lukasjuhrich) [inline sipa/model/hss/user.py]: A static `has_connection = True` would've achieved the same, but I don't care about cosmetics atm.

File: sipa/model/hss/schema.py [hunk 4]
```diff
@@ -27,6 +27,7 @@ class Account(db.Model):
     traffic_log = relationship('TrafficLog')
     traffic_quota_id = Column('traffic_quota', Integer, ForeignKey('traffic_quota.id'))
     # traffic_quota is available using a backref
+    use_cache = Column(Boolean, nullable=False)
```
Comment (lukasjuhrich) [inline sipa/model/hss/schema.py]: Use `default=False` for convenience in the tests.

File: sipa/model/hss/user.py (hunk 3 above)
Comment (Lodifice) [reply sipa/model/hss/user.py]: I agree with you, but _in theory_ the best practice would be to not delete accounts in Abe and mark the as closed in some sense. Then we would need a more sophisticated function here.

However, I've heard that Pycroft offers better solutions anyway ;)

**Comment by Lodifice** [comment]
@lukasjuhrich I don't have any problem with the revert either. The only intention for introducing it quickly was to make use of the fact that you have time for a merge window^^

**Pull request merged**